packages/credentials/src/types.ts

```diff
@@ -0,0 +1,79 @@
+
+/**
+ * @see {@link https://www.w3.org/TR/vc-data-model/#credentials | VC data model}
+ */
+export type VerifiableCredential = {
+  /** see {@link https://www.w3.org/TR/vc-data-model/#contexts | Contexts} */
+  '@context': ['https://www.w3.org/2018/credentials/v1', ...string[]];
+  credentialSubject: CredentialSubject | CredentialSubject[];
+  /** @see {@link https://www.w3.org/TR/vc-data-model/#identifiers | Identifiers} */
+  id?: string;
+  /** @see {@link IssuerType} */
+  issuer: Issuer;
+  /** @see {@link https://www.w3.org/TR/vc-data-model/#types | Types} */
+  type?: string[] | string;
+  /** @see {@link https://www.w3.org/TR/vc-data-model/#issuance-date | Issuance Date} */
+  issuanceDate: string;
+  /** @see {@link https://www.w3.org/TR/vc-data-model/#expiration | Expiration} */
+  expirationDate?: string;
+  /** @see {@link CredentialStatusReference} */
+  credentialStatus?: CredentialStatusReference;
+  [x: string]: any;
+}
+
+/**
+ * The issuer of a {@link VerifiableCredential}.
+ * The value of the issuer property must be either a URI or an object containing an `id` property.
+ * It is _recommended_ that the URI in the issuer or its id be one which, if de-referenced, results in a document
+ * containing machine-readable information about the issuer that can be used to verify the information expressed in the
+ * credential.
+ *
+ * See {@link https://www.w3.org/TR/vc-data-model/#issuer | Issuer data model}
+ */
+export type Issuer = {
+  /** @see {@link https://www.w3.org/TR/vc-data-model/#identifiers | Identifiers} */
+  id: string;
+  [x: string]: any;
+} | string
+
+/**
+ * The value of the `credentialSubject` property is defined as a set of objects that contain one or more properties that
+ * are each related to a subject of the verifiable credential. Each object MAY contain an id.
+ *
+ * See {@link https://www.w3.org/TR/vc-data-model/#credential-subject | Credential Subject}
+ */
+export type CredentialSubject = {
+  /** @see {@link https://www.w3.org/TR/vc-data-model/#identifiers | Identifiers} */
+  id?: string;
+  [x: string]: any;
```

Comment:
Tiny nit -- apologies for raising 🤓 

`[x: string]` convention is used throughout Veramo's lib but we've typically been using `[key: string]` or `[prop: string]` in Web5 SDK / DWN SDK.  Although to be fair, we've not been consistent.

Reply:
good point! fixed.